### PR TITLE
Reduce a bit more duplication on *_from_stream.

### PR DESCRIPTION
**Pull request by bobmcwhirter**

### CODE REVIEW

File: modules/ingestor/src/service/format.rs [hunk 1]
```diff
@@ -230,14 +230,10 @@ where
...
-                .in_scope(|| serde_json::from_reader(reader))
-                .map_err(Error::Json),
-            StreamFlavor::Yaml => info_span!("parse document")
-                .in_scope(|| serde_yml::from_reader(reader))
-                .map_err(Error::Yaml),
-        }
+        info_span!("parse document").in_scope(|| match flavor {
+            StreamFlavor::Yaml => serde_yml::from_reader(reader).map_err(Error::Yaml),
+            StreamFlavor::Json => serde_json::from_reader(reader).map_err(Error::Json),
+        })
```
Comment (jcrossley3) [inline modules/ingestor/src/service/format.rs:236]: I feel like the construction of the bridge should be a part of the parsing span, so will this work?

```rust
    tokio::task::spawn_blocking(move || {
        info_span!("parse document").in_scope(|| {
            let stream = pin!(stream);
            let stream = stream.map_err(|e| io::Error::new(io::ErrorKind::Other, format!("{e:?}")));
            let reader = SyncIoBridge::new(StreamReader::new(stream));
            match flavor {
                StreamFlavor::Json => serde_json::from_reader(reader).map_err(Error::Json),
                StreamFlavor::Yaml => serde_yml::from_reader(reader).map_err(Error::Yaml),
            }
        })
    })
    .await?
```

**Review by jcrossley3 — CHANGES_REQUESTED** [review]

**Review by jcrossley3 — APPROVED** [review]

File: modules/ingestor/src/service/format.rs (hunk 1 above)
Comment (jcrossley3) [reply modules/ingestor/src/service/format.rs:236]: Nevermind. I changed my mind.

Comment (bobmcwhirter) [reply modules/ingestor/src/service/format.rs:236]: Lemme try!

Comment (bobmcwhirter) [reply modules/ingestor/src/service/format.rs]: Seems to.

**Pull request merged**